lib/segment/src/index/hnsw_index/graph_layers_builder.rs

```diff
@@ -419,6 +420,7 @@ impl GraphLayersBuilder {
                             }
                         }
                     } else {
+                        search_context.nearest.prune_elements();
```

Comment:
Couldn't figure out a simple way to make this happen inside `iter()`. Do you have any ideas?

Reply:
I don't see too much problems with it

Reply:
Okay.